build.gradle.kts

```diff
@@ -212,16 +310,20 @@ subprojects {
             classifier = ""
             // no need to add analyzer.init.gradle in the jar since it will never be used from inside the plugin itself
             exclude("analyzer-init.gradle")
-            // Minimise the resulting uber-jars to ensure we don't have massive jars
-            minimize() {
-                // Sometimes minimisation takes away too much ... ensure we keep these.
-                exclude(dependency("com.fasterxml.jackson.core:.*:.*"))
-                exclude(dependency("org.commonjava.maven.ext:.*:.*"))
-                exclude(dependency("org.commonjava.maven.atlas:.*:.*"))
-                exclude(dependency("org.aeonbits.owner:.*:.*"))
-                exclude(dependency("org.slf4j:.*:.*"))
-                exclude(dependency("org.apache.maven:.*:.*"))
+
+            if (org.gradle.util.GradleVersion.current() >= org.gradle.util.GradleVersion.version("5.0")) {
+                // Minimise the resulting uber-jars to ensure we don't have massive jars
+                minimize {
+                    // Sometimes minimisation takes away too much ... ensure we keep these.
+                    exclude(dependency("com.fasterxml.jackson.core:.*:.*"))
+                    exclude(dependency("org.commonjava.maven.ext:.*:.*"))
+                    exclude(dependency("org.commonjava.maven.atlas:.*:.*"))
+                    exclude(dependency("org.aeonbits.owner:.*:.*"))
+                    exclude(dependency("org.slf4j:.*:.*"))
+                    exclude(dependency("org.apache.maven:.*:.*"))
```

Comment:
Good choice, keeps it simple. I would however modify the comment to explain why we only minimise on >= 5.0

Reply:
I added a comment. Also, nothing else seemed to work for me except to not run it at all (meaning `exclude()` had no effect).